subsys/bluetooth/host/conn.c

```diff
@@ -1702,11 +1713,19 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 				notify_connected(conn);
 			}
 
+			bt_conn_unref(conn);
+		} else if (old_state == BT_CONN_CONNECT_ADV) {
+			if (conn->err) {
+				notify_connected(conn);
+			}
```

Comment:
Notifying the application in this case doesn't seem right since we haven't exposed the connection to the application at this point. It would not even have a real address assigned to it.

Reply:
Yes, that is a good point. Removing connectable advertiser from here.

What do you think about BT_CONN_CONNECT_AUTO?
In the other PR (adv/scan addr fix) I notify the application that the auto conn has been stopped by RPA timeout.

With this change I now have another option and that is to restart the auto-conn, since the conn-params are in the connection object. But I'm not convinced it is the best option.
So it is either: 1. stop at RPA timeout. 2 toggle whitelist initiator at RPA timeout.
Possible: 3: add different timeout on auto-conn, same as direct-conn.
Thoughs?

Reply:
I didn't realise we stopped automatic connection creation due to RPA timeout. I don't think we should do that, i.e. we should restart whatever automatic connection requests were ongoing after RPA timeout.

Btw, did you notice that there are some Shippable failures?

Reply:
Yes, fixed shippable, needed to consider advertiser without peripheral.

At the moment, if the initiator is running when RPA timeout happen, we will fail to update the random address. I'm trying to fix it here: #21714
My current approach (int that PR) is to just disable the whitelist initiator, after having first made sure that the RPA timeout is restartet, so that the whitelist initiator will run for a full RPA period.
The reason for this approach was that I didn't have a copy of the conn params.

With this PR I have the conn param as they are in the conn-object. So I guess we should probably go with the restart approach? I think it needs an additional state variable to handle race-conditions between app stopping the initiator in the middle of the RPA timeout. Which is why I'm currently wondering if the simple stop approach is better.

Reply:
The automatic connection establishment is something that's supposed to continue over a long period of time, in most cases as long as the device is paired, so stopping it at RPA timeout doesn't seem right. Even though it might increase implementation complexity it'll reduce it on the app side, which is what should take priority (in addition to avoiding surprises in the API usage).

Btw, are you expecting #21714 to be reviewed & merged independently of this one, do you consider them to be mutually exclusive, or what?

Reply:
Agreed. Then we will toggle the whitelist initiator at RPA timeout.

I created #21714 first, but it currently depends on changes in this PR. So should be reviewed after this one has been merged and I have made the update to toggle the whitelist initiator.
I need the conn params to toggle the initiator, and the easiest solution to that is to use the conn params from the conn object, which is the dependency on this PR.